core/drivers/pmd.cc

```diff
@@ -324,16 +324,20 @@ CommandResponse PMDPort::Init(const bess::pb::PMDPortArg &arg) {
   return CommandSuccess();
 }
 
-int PMDPort::UpdateConf(const Conf &conf) {
+CommandResponse PMDPort::UpdateConf(const Conf &conf) {
   rte_eth_dev_stop(dpdk_port_id_);
 
   if (conf_.mtu != conf.mtu && conf.mtu != 0) {
+    if (conf.mtu > SNBUF_DATA || conf.mtu < ETHER_MIN_MTU) {
+      return CommandFailure(EINVAL, "mtu should be >= %d and <= %d",
+                            ETHER_MIN_MTU, SNBUF_DATA);
+    }
+
     int ret = rte_eth_dev_set_mtu(dpdk_port_id_, conf.mtu);
     if (ret == 0) {
-      conf_.mtu = conf_.mtu;
+      conf_.mtu = conf.mtu;
     } else {
-      LOG(WARNING) << "rte_eth_dev_set_mtu() failed: " << rte_strerror(-ret);
-      return ret;
+      return CommandFailure(-ret, "rte_eth_dev_set_mtu() failed");
```

Comment:
It would be helpful to include the output of `rte_strerror()` here, too. And shouldn't we keep the original `LOG(WARNING)`?

Reply:
In a case of RPC call (e.g., through `bessctl`), returning `CommandResponse` without warning is one of the common patterns used in this repo. `CommandFailure()` or `CommandSuccess()` will be returned to the RPC caller, so that the caller should log that result if necessary. (Refer the code structures in `besctl.cc` https://github.com/NetSys/bess/blob/9cfa4bb188ebde987b43a617fdbd096471888314/core/bessctl.cc#L1039)